src/main/java/io/axoniq/axonserver/connector/impl/FlowControlledBuffer.java

```diff
@@ -111,11 +118,36 @@ protected T tryTakeNow() {
      *
      * @param timeout  the duration to wait for an entry to become available in the buffer
      * @param timeUnit the {@link TimeUnit} used to specify the duration together with the {@code timeout}
-     * @return an entry of type {@code T} from this buffer if present, otherwise {@code null}
+     * @return an entry of type {@code T} from this buffer if present, otherwise {@code null}. Timeouts will result in
+     * null as well
      * @throws InterruptedException while waiting for an entry to be taken
      */
     protected T tryTake(long timeout, TimeUnit timeUnit) throws InterruptedException {
-        T taken = validate(buffer.poll(timeout, timeUnit), true);
+        try {
+            return tryTake(timeout, timeUnit, false);
+        } catch (TimeoutException e) {
+            return null;
```

Comment:
Thought: isn't it worth logging this scenario, perhaps just on debug?

Reply:
It's an impossible scenario, it's only there because java enforces the checked exceptions. The third parameter having false indicates it should never throw

Reply:
I changed the API, so the catch is no longer needed